docs/source/tutorials/vision/custom_task_tutorial.rst

```diff
@@ -0,0 +1,280 @@
+  train_ds = CocoSegmentDataset.get_download(root, train=True)
+  test_ds = CocoSegmentDataset.get_download(root, train=False)
+
+  def batch_collate(batch):
+      imgs, classes, masks = zip(*batch)
+      return list(imgs), list(classes), list(masks)
+
+  train_data_loader = DataLoader(
+      dataset=train_ds,
+      batch_size=32,
+      shuffle=False,
+      collate_fn=batch_collate
+  )
+
+  test_data_loader = DataLoader(
+      dataset=test_ds,
+      batch_size=32,
+      shuffle=False,
+      collate_fn=batch_collate
+  )
+
+
+Visualizing that our dataset was loaded correctly:
+
+.. code-block:: python
+
+  import numpy as np
+  import matplotlib.pyplot as plt
+
+  from torchvision.utils import draw_segmentation_masks
+  import torchvision.transforms.functional as F
+
+  def show(imgs):
+      if not isinstance(imgs, list):
+          imgs = [imgs]
+      fix, axs = plt.subplots(ncols=len(imgs), figsize=(20,20))
+      for i, img in enumerate(imgs):
+          img = img.detach()
+          img = F.to_pil_image(img)
+          axs[i].imshow(np.asarray(img))
+          axs[i].set(xticklabels=[], yticklabels=[], xticks=[], yticks=[])
+
+
+  imgs = [draw_segmentation_masks(train_ds[i][0], masks=train_ds[i][2], alpha=0.8)
+          for i in range(5)]
+
+  show(imgs)
+
+.. image :: /_static/custom-segment-tutorial.png
+  :width: 400
+  :alt: COCO128 segmentation visualize
+
+For the model we will use a pretrain model from `torchvision`
+
+.. code-block:: python
+
+  from torchvision.models.detection import fasterrcnn_resnet50_fpn
+
+  model = fasterrcnn_resnet50_fpn(pretrained=True, progress=False)
+  model = model.eval()
+
+
+Implement Custom `VisionData`
+=============================
+
+When we have our `DataLoader`s and model ready we can start creating our own implementation
+of `VisionData` class.
+
+.. code-block:: python
+
+  from typing import List, Iterable
+  from deepchecks.vision.vision_data import VisionData
+
+  class MyCustomSegmentationData(VisionData):
+      """Class for loading the COCO segmentation dataset."""
+
+      def get_classes(self, batch_labels) -> List[List[int]]:
+          """Return per label a list of classes (by id) in it."""
+          # The input `batch_labels` is the result of `batch_to_labels` function.
+          return batch_labels[0]
+
+      def batch_to_labels(self, batch):
+          """Extract from the batch only the labels."""
+          return batch[1], batch[2]
+
+      def infer_on_batch(self, batch, model, device):
+          """Infer on a batch of images."""
+          predictions = model.to(device)(batch[0])
+          return predictions
+
+      def batch_to_images(self, batch) -> Iterable[np.ndarray]:
+          """Convert the batch to a list of images as (H, W, C) 3D numpy array per image."""
+          return [tensor.numpy().transpose((1, 2, 0)) for tensor in batch[0]]
+
+Now we are able to run checks that are working only on the image data, since it's in the standard format of
```

Comment:
```suggestion
Now we are able to run checks that use only the image data, since it's in the standard Deepchecks format
```
Link to explanation of image format.